Taarafo.Core/Controllers/GroupsController.cs

```diff
@@ -75,5 +76,29 @@ public ActionResult<IQueryable<Group>> GetAllGroups()
                 return InternalServerError(groupServiceException.InnerException);
             }
         }
+
+        [HttpGet]
+        public async ValueTask<ActionResult<Group>> GetGroupByIdAsync(Guid groupId)
+        {
+            try
+            {
+                Group retrievedGroupById = 
+                    await this.groupService.RemoveGroupByIdAsync(groupId);
+
+                return Ok(retrievedGroupById);
+            }
+            catch (GroupValidationException groupValidationException)
```

Comment:
`catch (GoupValidationException groupValidationException)
                when (groupValidationException.InnerException is NotFoundGroupException)
            {
                return NotFound(groupValidationException.InnerException);
            }`   How is it return Not Found error without this code?

Reply:
I have added this exception. You can check it out